### ISSUE TITLE
Menus not working

### ISSUE THREAD

**Issue by char0n** [CodinGame/monaco-vscode-api#267]
Hello again,

I've noticed that the menus in monaco-editor + monaco-vscode-api stopped working at some point of the time. I couldn't pinpoint exact version where it stopped working.

**This is how it used to work:**

https://github.com/CodinGame/monaco-vscode-api/assets/193286/2dae49e2-e0c7-45b4-8659-fcf807fe180f

**This is how it behaves when not working:**

https://github.com/CodinGame/monaco-vscode-api/assets/193286/6e4d909b-58e7-4060-a804-10c36ef61c5f

---

The non working menu is missing `Go to Symbol` and clicking on `Command Palette` just does nothing. Do you have any idea how to make it work again?

Thank you very much!

**Comment by char0n** [comment]
@CGNonofr nope, I've tried before creating the issue. 

I don't use any overrides. I use just simple service and extensions initialization:

https://github.com/swagger-api/swagger-editor/blob/d5353177f15544f2592cbbd721cda182f2422506/src/plugins/editor-monaco/after-load.js#L38

**Comment by char0n** [comment]
Did some further digging and it seems like the `Go to symbol...` menu item is displayed on certain `precondition`:

```
EditorContextKeys.hasDocumentSymbolProvider
```

which is created as:

```
_languageFeaturesService.documentSymbolProvider.has(model)
```

**Comment by char0n** [comment]
> are you able to reproduce the issue in the demo here? (aka https://codingame.github.io/monaco-vscode-api/)

@CGNonofr I can reproduce. `CTRL+SHIFT+O` does show the symbols, but the menu is not displaying `Go to symbol...` menu item.

**Comment by char0n** [comment]
Related to `Command palette` dialog I've managed to get an error out of the editor/vscode by pressing `CTRL+SHIFT+p`.

![image](https://github.com/CodinGame/monaco-vscode-api/assets/193286/70e8b4eb-ed85-471f-9573-cf07c4ce7028)

**Comment by CGNonofr** [comment]
It seems that the `Go to symbol...` menu is a monaco feature only (which doesn't exist in VSCode) so this lib removes it

regarding the preserveInput error, there is a missing config key that is loaded only when the view or keybindings service overrides are used, fixed in https://github.com/CodinGame/monaco-vscode-api/pull/269/commits/8d56e2af4b68551aebf206057591304fdb13f714 (open MR)

**Comment by char0n** [comment]
> It seems that the Go to symbol... menu is a monaco feature only (which doesn't exist in VSCode) so this lib removes it

Would it do any harm to keep it there?

> regarding the preserveInput error, there is a missing config key that is loaded only when the view or keybindings service overrides are used, fixed in https://github.com/CodinGame/monaco-vscode-api/commit/8d56e2af4b68551aebf206057591304fdb13f714 (open MR)

Thanks a lot!

**Comment by char0n** [comment]
> Probably not, but it's mixed with everything else which is removed for valid reasons

Trying to create a custom menu item, to see if I can add it easily back

**Comment by char0n** [comment]
Managed to get it working with the following IActionDescriptor:

```js
import * as monaco from 'monaco-editor';
import { StandaloneServices, IQuickInputService } from 'vscode/services';

const goToSymbolActionDescriptor = {
  id: 'editor.action.quickOutline',
  label: 'Go to Symbol...',
  precondition: 'editorHasDocumentSymbolProvider',
  contextMenuGroupId: 'navigation',
  contextMenuOrder: 3,
  keybindings: [monaco.KeyMod.CtrlCmd | monaco.KeyMod.Shift | monaco.KeyCode.KeyO], // eslint-disable-line no-bitwise
  async run() {
    StandaloneServices.get(IQuickInputService).quickAccess.show('@');
  },
};

export default goToSymbolActionDescriptor;
```

**Comment by char0n** [comment]
> regarding the preserveInput error, there is a missing config key that is loaded only when the view or keybindings service overrides are used, fixed in https://github.com/CodinGame/monaco-vscode-api/commit/8d56e2af4b68551aebf206057591304fdb13f714 (open MR)

Before the fix is released, the workaround is to explicitly use following import:

```js
import 'vscode/vscode/vs/workbench/browser/workbench.contribution';
```

@CGNonofr thanks for all the support!

**Comment by char0n** [comment]
Resolved in @codingame/monaco-vscode-api@=1.83.15
